its/src/test/java/com/sonar/it/scanner/msbuild/ScannerMSBuildTest.java

```diff
@@ -225,12 +226,30 @@ private void assertLineCountForProjectUnderTest(String projectKey) {
   @Test
   public void testExcludedAndTest_AnalyzeTestProject() throws Exception {
     int expectedTestProjectIssues = isTestProjectSupported() ? 1 : 0;
-    testExcludedAndTest(false, expectedTestProjectIssues);
+    String token = TestUtils.getNewToken(ORCHESTRATOR);
+    Path projectDir = TestUtils.projectDir(temp, "ExcludedTest");
+    ScannerForMSBuild build = TestUtils.newScannerBegin(ORCHESTRATOR, "ExcludedTest_False", projectDir, token, ScannerClassifier.NET_FRAMEWORK_46)
+      // don't exclude test projects
+      .setProperty("sonar.dotnet.excludeTestProjects", "false");
+    testExcludedAndTest(build, "ExcludedTest_False", projectDir, token, expectedTestProjectIssues);
   }
 
   @Test
   public void testExcludedAndTest_ExcludeTestProject() throws Exception {
-    testExcludedAndTest(true, 0);
+    String token = TestUtils.getNewToken(ORCHESTRATOR);
+    Path projectDir = TestUtils.projectDir(temp, "ExcludedTest");
+    ScannerForMSBuild build = TestUtils.newScannerBegin(ORCHESTRATOR, "ExcludedTest_True", projectDir, token, ScannerClassifier.NET_FRAMEWORK_46)
+      // exclude test projects
+      .setProperty("sonar.dotnet.excludeTestProjects", "true");
+    testExcludedAndTest(build, "ExcludedTest_True", projectDir, token, 0);
+  }
+
+  @Test
```

Comment:
[Education] What is the difference between this test and `testExcludedAndTest_ExcludeTestProject` if now the `SONARQUBE_SCANNER_PARAMS` is set unconditionally?

Reply:
oops... good catch!

`     EnvironmentVariable sonarQubeScannerParams = new EnvironmentVariable("SONARQUBE_SCANNER_PARAMS", "{\"sonar.dotnet.excludeTestProjects\":\"true\",\"sonar.verbose\":\"true\"}");` should be created here, not in `testExcludedAndTest`

I guess things worked because the property taked precedence somehow in `testExcludedAndTest_AnalyzeTestProject`